src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,38 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<Machine> machines = new ArrayList<>();
+        if (type == Bulldozer.class) {
+            machines.addAll(new BulldozerProducer().get());
+        } else if (type == Excavator.class) {
+            machines.addAll(new ExcavatorProducer().get());
+        } else if (type == Truck.class) {
+            machines.addAll(new TruckProducer().get());
+        }
+        return machines;
```

Comment:
Instead of returning null or a specific type list in `getAll()` when no producer has been found, it's better to return an empty list to avoid potential NullPointerException.